lib/rouge/lexers/sparql.rb

```diff
@@ -41,16 +41,17 @@ def self.keywords
         rule %r('''), Str::Single, :string_single_literal
         rule %r('), Str::Single, :string_single
 
-        rule %r([$?]\w+), Name::Variable
-        rule %r((\w*:)(\w+)?) do |m|
+        rule %r([$?][[:word:]]+), Name::Variable
+        rule %r((([[:word:]]|[_-])*)(:)(([[:word:]]|[_-])+)?) do |m|
```

Comment:
**suggestion:** We can simplify this regex a bit more. 

- There seems to be some unexpected capture groups 
  ```rb
  [2] pry(main)> "podmínky-užití:databáze-chráněná-zvláštními-právy".scan(regex)
  => [["podmínky-užití", "í", ":", "databáze-chráněná-zvláštními-právy", "y"]]
  ```
- `[:word:]` also matches punctuation including underscore `_` 

```suggestion
        rule %r(([[:word:]-]*)(:)([[:word:]-]+)?) do |m|
```

And that means we end up with exactly 3 capture groups and can adjust the matches accordingly. What do you think?